src/gui/properties/speedplotview.h

```diff
@@ -112,21 +130,14 @@ class SpeedPlotView final : public QGraphicsView
     };
 
     quint64 maxYValue();
-    boost::circular_buffer<PointData> &getCurrentData();
-
-    boost::circular_buffer<PointData> m_data5Min;
-    boost::circular_buffer<PointData> m_data30Min;
-    boost::circular_buffer<PointData> m_data6Hour;
-    boost::circular_buffer<PointData> m_data12Hour;
-    boost::circular_buffer<PointData> m_data24Hour;
-    boost::circular_buffer<PointData> *m_currentData;
+    const DataCircularBuffer &getCurrentData() const;
+
+    Averager m_averager5Min;
     Averager m_averager30Min;
     Averager m_averager6Hour;
     Averager m_averager12Hour;
     Averager m_averager24Hour;
+    Averager *m_currentData;
```

Comment:
Let's call things more explicitly. Otherwise, we have in the code a bunch of "data", "current data", etc., which actually refer to completely different things.
So maybe call it `m_currentAverager`?